### PR TITLE
Use an enum instead of bool to aid readability

### PR DESCRIPTION
**Pull request by akhi3030**
I find the use of `bool` here a bit limiting as I can't remember what `true` and `false` mean.  Should we replace them with an `enum`?

### CODE REVIEW

**Comment by akhi3030** [comment]
@mina86 : if you think this is a reasonable change, then I'll finish up the patch and submit for proper review.

**Comment by mina86** [comment]
LGTM

File: core/store/src/config.rs [hunk 1]
```diff
@@ -170,13 +172,19 @@ impl<'a> StoreOpener<'a> {
...
+            match self.mode {
+                Mode::ReadOnly => {
+                    tracing::error!(target: "near", path=%self.path.display(), "Database does not exist");
+                    panic!("Failed to open non-existent the database");
+                }
+                Mode::ReadWrite => {
+                    tracing::info!(target: "near", path=%self.path.display(), "Creating new RocksDB database");
+                }
+            }
         }
```
Comment (mina86) [inline core/store/src/config.rs:185]: It’d be easire to do `if matches!(Mode::ReadOnly, self.mode)`.

File: core/store/src/db.rs [hunk 2]
```diff
@@ -171,22 +171,23 @@ fn ensure_max_open_files_limit(max_open_files: u32) -> Result<(), DBError> {
     }
 }
 
+#[derive(Clone)]
```
Comment (mina86) [inline core/store/src/db.rs:174]: `Copy` as well and then you don’t need to `clone`.

File: neard/src/cli.rs [hunk 3]
```diff
@@ -70,7 +70,8 @@ impl NeardCmd {
             NeardSubCommand::Run(cmd) => cmd.run(&home_dir, genesis_validation, runtime),
 
             NeardSubCommand::StateViewer(cmd) => {
-                cmd.subcmd.run(&home_dir, genesis_validation, cmd.readwrite);
+                let mode = if cmd.readwrite { Mode::ReadWrite } else { Mode::ReadOnly };
```
[the remark this answers is not in the document]
Comment (mina86) [reply neard/src/cli.rs:73]: It’s not messing around with clap IMO.  This looks good to me.

**Review by mina86 — APPROVED** [review]

**Pull request merged**